### PR TITLE
[Refactor] Refactor the transformation from image to camera coordinates

### PR DESCRIPTION
**Pull request by Tai-Wang**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation
Please describe the motivation of this PR and the goal you want to achieve through this PR.
Move the transformation from image to camera in the FCOS3D to mmdet3d/core such that it is more convenient for other classes or functions to use it.

## Modification
Please briefly describe what modification is made in this PR.
Add deprecation warning for the static method pts2Dto3D in the FCOS3D.
Rename it as points_img2cam and move it into mmdet3d/core.

## BC-breaking (Optional)
Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.
No.

## Use cases (Optional)
If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet3d/core/bbox/structures/utils.py [hunk 1]
```diff
@@ -213,6 +213,40 @@ def points_cam2img(points_3d, proj_mat, with_depth=False):
...
+    points2D = points[:, :2]
+    depths = points[:, 2].view(-1, 1)
+    unnorm_points2D = torch.cat([points2D * depths, depths], dim=1)
+
+    viewpad = torch.eye(4, dtype=points2D.dtype, device=points2D.device)
+    viewpad[:view.shape[0], :view.shape[1]] = points2D.new_tensor(view)
+    inv_viewpad = torch.inverse(viewpad).transpose(0, 1)
+
+    # Do operation in homogenous coordinates.
+    nbr_points = unnorm_points2D.shape[0]
```
Comment (ZwwWayne) [inline mmdet3d/core/bbox/structures/utils.py:242]: nbr

Comment (ZwwWayne) [reply mmdet3d/core/bbox/structures/utils.py:242]: unnorm -> unnormed

File: mmdet3d/core/bbox/structures/utils.py [hunk 2]
```diff
@@ -213,6 +213,40 @@ def points_cam2img(points_3d, proj_mat, with_depth=False):
...
+
+    Returns:
+        torch.Tensor: points in 3D space. [N, 3],
+            3 corresponds with x, y, z in 3D space.
+    """
+    assert view.shape[0] <= 4
+    assert view.shape[1] <= 4
+    assert points.shape[1] == 3
+
+    points2D = points[:, :2]
```
Comment (ZwwWayne) [inline mmdet3d/core/bbox/structures/utils.py:233]: points_xy

File: mmdet3d/core/bbox/structures/utils.py [hunk 3]
```diff
@@ -213,6 +213,40 @@ def points_cam2img(points_3d, proj_mat, with_depth=False):
     return point_2d_res
 
 
+def points_img2cam(points, view):
+    """Project points in image coordinates to camera coordinates.
+
+    Args:
+        points (torch.Tensor): 2.5D points in 2D images, [N, 3],
+            3 corresponds with x, y in the image and depth.
+        view (np.ndarray): Camera instrinsic matrix. The shape can be [3, 3],
```
Comment (ZwwWayne) [inline mmdet3d/core/bbox/structures/utils.py:222]: cam2img might be better

**Comment by ZwwWayne** [comment]
Need unit tests

**Comment by Tai-Wang** [comment]
> Need unit tests

Comments resolved.

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**